projects/observability/src/shared/dashboard/data/graphql/explore/explore-result.ts

```diff
@@ -69,20 +69,24 @@ export class ExploreResult {
   }
 
   private extractMetricAggregationForSpec(spec: ExploreSpecification): MetricAggregationData {
-    return this.resultToMetricAggregationDataData(this.resultsContainingSpec(spec)?.[0]!, spec);
+    return this.resultToMetricAggregationDataData(this.resultsContainingSpec(spec)?.[0], spec);
   }
 
   private extractTimeseriesForSpec(spec: ExploreSpecification): MetricTimeseriesInterval[] {
     return this.resultsToTimeseriesIntervals(this.resultsContainingSpec(spec), spec);
   }
 
   private resultToMetricAggregationDataData(
-    result: GraphQlExploreResult,
+    result: GraphQlExploreResult | undefined,
     spec: ExploreSpecification
   ): MetricAggregationData {
     return {
       key: spec.name,
-      value: result[spec.resultAlias()].value as number
+      ...(result
+        ? {
+            value: result?.[spec.resultAlias()].value as number
```

Comment:
if we sent value as undefined, what would happen? It'd read a lot cleaner if we can remove the ternary and typically there wouldn't be much difference between `{value: undefined}` and `{}`